types/index.d.ts

```diff
@@ -183,13 +183,13 @@ declare namespace fastifyOauth2 {
         
         revokeAllToken(revokeToken: Token, httpOptions: Object | undefined): Promise<void>;
 
-        userinfo(tokenSetOrToken: Token | string): Promise<Object>;
+        userinfo<TUserInfo>(tokenSetOrToken: Token | string): Promise<TUserInfo>;
```

Comment:
Should not we set the default?

```suggestion
        userinfo<TUserInfo = Object>(tokenSetOrToken: Token | string): Promise<TUserInfo>;
```

Reply:
Why so? the generic already serves as a default value